tests/CsvFileTest.php

```diff
@@ -0,0 +1,356 @@
+    }
+
+    /**
+     * @expectedException \Keboola\Csv\Exception
+     * @expectedExceptionMessage Cannot open file
+     */
+    public function testExceptionShouldBeThrownOnNotExistingFile()
+    {
+        $csv = new CsvFile(__DIR__ . '/something.csv');
+        $csv->getHeader();
+    }
+
+    public function testColumnsCount()
+    {
+        $csv = new CsvFile(__DIR__ . '/data/test-input.csv');
+
+        self::assertEquals(9, $csv->getColumnsCount());
+    }
+
+    /**
+     * @dataProvider validCsvFiles
+     * @param string $fileName
+     * @param string $delimiter
+     */
+    public function testRead($fileName, $delimiter)
+    {
+        $csvFile = new CsvFile(__DIR__ . '/data/' . $fileName, $delimiter, '"');
+
+        $expected = [
+            "id",
+            "idAccount",
+            "date",
+            "totalFollowers",
+            "followers",
+            "totalStatuses",
+            "statuses",
+            "kloutScore",
+            "timestamp",
+        ];
+        self::assertEquals($expected, $csvFile->getHeader());
+    }
+
+    public function validCsvFiles()
+    {
+        return [
+            ['test-input.csv', ','],
+            ['test-input.win.csv', ','],
+            ['test-input.tabs.csv', "\t"],
+            ['test-input.tabs.csv', "	"],
+        ];
+    }
+
+    public function testParse()
+    {
+        $csvFile = new CsvFile(__DIR__ . '/data/escaping.csv', ",", '"');
+
+        $rows = [];
+        foreach ($csvFile as $row) {
+            $rows[] = $row;
+        }
+
+        $expected = [
+            [
+                'col1', 'col2',
+            ],
+            [
+                'line without enclosure', 'second column',
+            ],
+            [
+                'enclosure " in column', 'hello \\',
+            ],
+            [
+                'line with enclosure', 'second column',
+            ],
+            [
+                'column with enclosure ", and comma inside text', 'second column enclosure in text "',
+            ],
+            [
+                "columns with\nnew line", "columns with\ttab",
+            ],
+            [
+                "Columns with WINDOWS\r\nnew line", "second",
+            ],
+            [
+                'column with \n \t \\\\', 'second col',
+            ],
+        ];
+
+        self::assertEquals($expected, $rows);
+    }
+
+
+    public function testEmptyHeader()
+    {
+        $csvFile = new CsvFile(__DIR__ . '/data/test-input.empty.csv', ',', '"');
+
+        self::assertEquals([], $csvFile->getHeader());
+    }
+
+    /**
+     * @dataProvider invalidDelimiters
+     * @expectedException \Keboola\Csv\InvalidArgumentException
+     * @param string $delimiter
+     */
+    public function testInvalidDelimiterShouldThrowException($delimiter)
+    {
+        new CsvFile(__DIR__ . '/data/test-input.csv', $delimiter);
+    }
+
+    public function invalidDelimiters()
```

Comment:
nemáme nějaký standard na to jak pojmenovávat dataProvidery? Např. `provideStuffForMethod()`?

Reply:
nemame